UnityProject/Assets/Scripts/Machines/QuantumPad.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+using System.Linq;
+using Gateway;
+
+public class QuantumPad : NetworkBehaviour, ICheckedInteractable<HandApply>
+{
+	public QuantumPad connectedPad;
+
+	/// <summary>
+	/// Detects players/objects on itself every 1 second.
+	/// </summary>
+	public bool passiveDetect;
+
+	/// <summary>
+	/// Where should this pad drop you on the next pad?
+	/// </summary>
+	public PadDirection padDirection = PadDirection.OnTop;
+
+	/// <summary>
+	/// If you dont want the link to be changed.
+	/// </summary>
+	public bool disallowLinkChange;
+
+	public string messageOnTravelToThis;
+
+	private RegisterTile registerTile;
+
+	private Matrix Matrix => registerTile.Matrix;
+
+	private Vector3 travelCoord;
+
+	private SpriteHandler spriteHandler;
+
+	private bool doingAnimation;
+
+	/// <summary>
+	/// Temp until shuttle landings possible
+	/// </summary>
+	public bool IsLavaLandBase1;
+
+	/// <summary>
+	/// Temp until shuttle landings possible
+	/// </summary>
+	public bool IsLavaLandBase1Connector;
+
+	/// <summary>
+	/// Temp until shuttle landings possible
+	/// </summary>
+	public bool IsLavaLandBase2;
+
+	/// <summary>
+	/// Temp until shuttle landings possible
+	/// </summary>
+	public bool IsLavaLandBase2Connector;
+
+	[Server]
+	private void ServerSync(bool newVar)
+	{
+		doingAnimation = newVar;
+	}
+
+	private void Awake()
+	{
+		registerTile = GetComponent<RegisterTile>();
+		spriteHandler = GetComponentInChildren<SpriteHandler>();
+		spriteHandler.ChangeSprite(0);
+	}
+
+	private void Start()
+	{
+		//temp stuff
+
+		if (IsLavaLandBase1)
+		{
+			LavaLandManager.Instance.LavaLandBase1 = this;
+		}
+
+		if (IsLavaLandBase2)
+		{
+			LavaLandManager.Instance.LavaLandBase2 = this;
+		}
+
+		if (IsLavaLandBase1Connector)
+		{
+			LavaLandManager.Instance.LavaLandBase1Connector = this;
+		}
+
+		if (IsLavaLandBase2Connector)
+		{
+			LavaLandManager.Instance.LavaLandBase2Connector = this;
+		}
+	}
+
+	private void OnEnable()
+	{
+		if (!passiveDetect) return;
+		if (!CustomNetworkManager.IsServer) return;
+
+		UpdateManager.Add(ServerDetectObjectsOnTile, 1f);
+	}
+
+	private void OnDisable()
+	{
+		UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, ServerDetectObjectsOnTile);
+	}
+
+	public bool WillInteract(HandApply interaction, NetworkSide side)
+	{
+		if (!DefaultWillInteract.Default(interaction, side)) return false;
+
+		if (Validations.IsTarget(gameObject, interaction)) return true;
+
+		return false;
+	}
+
+	public void ServerPerformInteraction(HandApply interaction)
+	{
+		ServerDetectObjectsOnTile();
+	}
+
+	public void ServerDetectObjectsOnTile()
+	{
+		if (connectedPad == null) return;
+
+		if (!doingAnimation && !passiveDetect)
+		{
+			ServerSync(true);
+
+			StartCoroutine(ServerAnimation());
+		}
+
+		travelCoord = connectedPad.registerTile.WorldPositionServer;
+
+		switch (padDirection)
```

Comment:
Codacy found an issue: [Add a 'default' clause to this 'switch' statement.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6280689)